applications/OptimizationApplication/custom_utilities/response/mass_response_utils.cpp

```diff
@@ -0,0 +1,271 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Reza Najian Asl
+//                   Suneth Warnakulasuriya
+//
+
+// #pragma once
+
+// System includes
+#include "includes/define.h"
+
+// Project includes
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/element_size_calculator.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "optimization_application_variables.h"
+
+// Include base h
+#include "mass_response_utils.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+bool MassResponseUtils::HasVariableInProperties(
+    const ModelPart& rModelPart,
+    const Variable<double>& rVariable)
+{
+    KRATOS_TRY
+
+    bool local_has_variable = false;
+    if (rModelPart.NumberOfElements() > 0) {
+        const auto& r_properties = rModelPart.Elements().front().GetProperties();
+        local_has_variable = r_properties.Has(rVariable);
+    }
+    return rModelPart.GetCommunicator().GetDataCommunicator().OrReduceAll(local_has_variable);
+
+    KRATOS_CATCH("");
+}
+
+double MassResponseUtils::CalculateMass(const ModelPart& rModelPart)
+{
+    KRATOS_TRY
+
+    if (rModelPart.GetCommunicator().GlobalNumberOfElements() == 0) {
+        return 0.0;
+    }
+
+    KRATOS_ERROR_IF_NOT(HasVariableInProperties(rModelPart, DENSITY))
+        << "DENSITY is not found in element properties of " << rModelPart.FullName() << ".\n";
+
+    KRATOS_ERROR_IF(HasVariableInProperties(rModelPart, THICKNESS) && HasVariableInProperties(rModelPart, CROSS_AREA))
```

Comment:
I would not, because solid elements will not have both. Only shell elements (or plate) will have `THICKNESS`, beam (or truss) elements will have `CROSS_AREA`.